src/main.rs

```diff
@@ -42,12 +44,13 @@ fn main() {
 pub fn run_command(ctx: &Context, command: Command) -> Result<()> {
     // We always try to acquire the mutex but it is only strictly necessary for
     // the lock and source commands.
-    let _guard = match acquire_mutex(ctx, ctx.config_dir()) {
-        Ok(g) => Some(g),
-        Err(_) if !matches!(command, Command::Lock | Command::Source) => None,
-        Err(err) => {
-            return Err(err).context("failed to acquire lock on config directory");
+    let file_mutex = match acquire_mutex(ctx.config_dir()) {
```

Comment:
This pattern doesn't release the lock in the err case or if anything panics. `fmutex` had an RAII pattern which ensured this. Either we need to build that or we should use a different library.